bsp/espressif/esp/src/hals/esp32-c3/gpio.zig

```diff
@@ -0,0 +1,148 @@
+const std = @import("std");
+const microzig = @import("microzig");
+const peripherals = microzig.chip.peripherals;
+const IO_MUX = peripherals.IO_MUX;
+const GPIO = peripherals.GPIO;
+
+pub const Level = enum(u1) {
+    low = 0,
+    high = 1,
+};
+
+pub const DriveStrength = enum(u2) {
+    @"5mA" = 0,
+    @"10mA" = 1,
+    @"20mA" = 2,
+    @"40mA" = 3,
+};
+
+pub const Pin = struct {
+    number: u5,
+
+    pub const Config = struct {
+        input_enable: bool = false,
+        output_enable: bool = false,
+        open_drain: bool = false,
+        pullup_enable: bool = false,
+        pulldown_enable: bool = false,
+        input_filter_enable: bool = false,
+        output_invert: bool = false,
+        drive_strength: DriveStrength = DriveStrength.@"5mA",
+    };
+
+    pub fn init(number: u5, config: Config) Pin {
```

Comment:
I would split this out into `from_instance_number(number: u5) Pin` and then an `init(self: Pin) void`.

Also I'm not committed to this, but at least the current standard terminology in MicroZig for these types of functions would be `apply()` and `reset()` (instead of `deinit()`):
https://github.com/ZigEmbeddedGroup/.github/blob/main/GUIDELINES.md#terminology